maintenance/troubleshoot/vpp/tracing-packets.md

```diff
@@ -0,0 +1,272 @@
+  tun4
+  00000000: 450002b056fd40003f0625650b6000010b00a68501bb8540a1f93599818eb1c1
+  00000020: 801801f620c700000101080a3f906c98fbaaba031703030277413d39b97817c1
+  00000040: 41392fdbe0e9d4886849851476cdb8986362ee2f789bfefd8a5c106c898d1309
+  00000060: 4f8f8cb89159d99e986813a48d91334930eb5eb10ca4248c
+00:09:46:518881: tun4-tx
+    buffer 0x24cf615: current data 0, length 688, buffer-pool 1, ref-count 1, totlen-nifb 0, trace handle 0x1000000
+  ipv4 tcp hdr-sz 52 l2-hdr-offset 0 l3-hdr-offset 0 l4-hdr-offset 20 l4-hdr-sz 32
+  0x0b60: 40:00:3f:06:25:65 -> 45:00:02:b0:56:fd
+
+Packet 2
+[...]
+````
+
+#### Traffic from the phy
+
+If you want to capture traffic coming from the physical NIC, you should use `trace add` but with a different source node a.k.a `dpdk-input` `af-packet-input` `af_xdp-input` `avf-input` instead of `virtio-input`. 
+
+`show run` should give you a hint of the `X-input` node you want to trace from.
+
+````
+vpp# show run
+Thread 1 vpp_wk_0 (lcore 25)
+Time 1.9, 10 sec internal node vector rate 1.05 loops/sec 1074819.68
+  vector rates in 7.5356e0, out 7.5356e0, drop 0.0000e0, punt 0.0000e0
+             Name                 State         Calls          Vectors        Suspends         Clocks       Vectors/Call
+avf-input                        polling           2233530               0               0          8.24e1            0.00
+ip4-cnat-snat                    active                  1               1               0          5.35e3            1.00
+ip4-cnat-tx                      active                 14              15               0          1.18e3            1.07
+[...]
+
+# Here we seem to want to use trace add avf-input 200
+````
+
+Same as with traffic from a container, you can use
+````bash
+# Make sure that the trace buffer is clean in VPP
+calivppctl vppctl NODENAME clear trace
+# Add a trace from the virtio-input input-node
+calivppctl vppctl NODENAME trace add avf-input 500
+# generate some traffic
+calivppctl vppctl NODENAME show trace max 500 > somefile
+# Grep for your IPs
+cat somefile | grep '1.2.3.4 -> 5.6.7.8' -A40 -B40
+````
+
+#### With wireshark
+
+Alternatively to the trace, you can do a capture and analyze it inside wireshark. You can do this with
+````
+vpp# pcap dispatch trace on max 1000 file vppcapture buffer-trace dpdk-input 1000
+vpp# pcap dispatch trace off
+````
+This will generate a file named `/tmp/vppcapture`
+
+Then on your host run:
+````bash
+calivppctl sh vpp NODENAME
+root@server:~# mv /tmp/vppcapture /run/vpp/
+root@server:~# exit
+# The file should now be at /run/vpp/vppcapture on your host 'NODENAME'
+````
+
+You can then `scp vppcapture` to your machine and open it with wireshark
+[More info about this here](https://haryachyy.wordpress.com/2019/09/29/learning-vpp-trace-with-wireshark/)
+
+### Traffic received in the pods
+
+In order to inspect traffic actually received by the pods if you have `tcpdump` installed in the pod and simply `tcpdump -ni eth0` inside the pod.
+
```

Comment:
To inspect traffic actually received by the pods (if `tcpdump` installed in the pod), simply `tcpdump -ni eth0` inside the pod.